### PR TITLE
fix `sync-blog` script, update to next 12

### PR DESCRIPTION
**Pull request by B2o5T**
fix sync-blog scripts, convert `check-links` to ts, update to next 12 and make compatible with `next-optimized-images`

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,67 +1,84 @@
...
-        },
-      ],
-      webpack5: false,
-      webpack(config, { dev, isServer }) {
-        if (!dev && isServer) {
-          const originalEntry = config.entry;
+      config.entry = async () => {
+        const entries = { ...(await originalEntry()) };
+
+        entries['./lib/build.ts'] = './lib/build.ts';
```
Comment (B2o5T) [inline next.config.js:33]: Everything looks fine, but deployment to Vercel fails with the following error
```text
HookWebpackError: File /vercel/path1/.next/server/lib/build.ts.js does not exist.
```
I think it related to this line

File: blog-sync.mjs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// Can't understand how to convert this file to ts 🤷‍
```
Comment (dotansimha) [inline blog-sync.mjs:1]: Yeah, agree. 

@kamilkisiela we talked about the same thing yesterday, right? writing a TS file and use ESM

File: next.config.js (hunk 1 above)
Comment (dotansimha) [reply next.config.js:33]: Super weird, is it something with our setup that causes that?

**Review by dotansimha — APPROVED** [review]

File: next.config.js (hunk 1 above)
Comment (kamilkisiela) [reply next.config.js:33]: let's try without it, it's for RSS and sitemap.xml but we run this script with ts-node right after `next build`. Not sure why it's here

File: blog-sync.mjs (hunk 2 above)
Comment (kamilkisiela) [reply blog-sync.mjs:1]: I tried to make it work but failed :)

Comment (kamilkisiela) [reply blog-sync.mjs:1]: I think it's fine to run `blog-sync.mjs` with `node` and turn `authors.js` to `authors.ts` + add `ts-node/register` (with `"module": "esnext"`).

**Comment by kamilkisiela** [comment]
Closes #538

**Comment by kamilkisiela** [comment]
Ok, I updated image optimization to the canary version, should no longer use binaries but webasembly instead. MDX does not work but I'll take a look tomorrow.

**Pull request merged**